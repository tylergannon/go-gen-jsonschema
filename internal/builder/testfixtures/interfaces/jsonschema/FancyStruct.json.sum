8400c1c239de0e1f