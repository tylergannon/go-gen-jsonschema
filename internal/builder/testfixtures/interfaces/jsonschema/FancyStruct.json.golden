{
  "type": "object",
  "description": "Make this look pretty interesting.",
  "properties": {
    "enumVal": {
      "type": "array",
      "description": "A list of enumVals that can be really meaningful when used correctly.",
      "items": {
        "type": "string",
        "description": "Overall description for MyEnumType.\n\nval1: \nThe first possible item\n\nval2: \nUse this one second\n\nval3: \nUse this one third\n\nval4: \nFourth option.",
        "enum": [
          "val1",
          "val2",
          "val3",
          "val4"
        ]
      }
    },
    "iface": {
      "anyOf": [
        {
          "type": "object",
          "description": "Put this down when you feel really great about life.",
          "properties": {
            "!type": {
              "type": "string",
              "const": "TestInterface1"
            },
            "field1": {
              "type": "string",
              "description": "obvious"
            },
            "field2": {
              "type": "string",
              "description": "oblivious"
            },
            "field3": {
              "type": "integer",
              "description": "obsequious"
            }
          },
          "required": [
            "!type",
            "field1",
            "field2",
            "field3"
          ]
        },
        {
          "type": "object",
          "description": "This is seriously silly, don't you imagine so?",
          "properties": {
            "!type": {
              "type": "string",
              "const": "TestInterface2"
            },
            "fork3": {
              "type": "integer"
            },
            "fork4": {
              "type": "integer"
            },
            "fork5": {
              "type": "integer"
            }
          },
          "required": [
            "!type",
            "fork3",
            "fork4",
            "fork5"
          ]
        },
        {
          "type": "object",
          "properties": {
            "!type": {
              "type": "string",
              "const": "PointerToTestInterface"
            },
            "fork99": {
              "type": "integer"
            },
            "fork10": {
              "type": "integer"
            },
            "fork11": {
              "type": "integer"
            }
          },
          "required": [
            "!type",
            "fork99",
            "fork10",
            "fork11"
          ]
        }
      ]
    },
    "Details": {
      "type": "array",
      "description": "Here are the details.  Make sure you fill them out.",
      "items": {
        "type": "object",
        "properties": {
          "Foo": {
            "type": "string",
            "description": "Highly interesting stuff regarding Foo and Bar."
          },
          "Bar": {
            "type": "string",
            "description": "Highly interesting stuff regarding Foo and Bar."
          },
          "enumVal": {
            "type": "string",
            "description": "Overall description for MyEnumType.\n\nval1: \nThe first possible item\n\nval2: \nUse this one second\n\nval3: \nUse this one third\n\nval4: \nFourth option.",
            "enum": [
              "val1",
              "val2",
              "val3",
              "val4"
            ]
          }
        },
        "required": [
          "Foo",
          "Bar",
          "enumVal"
        ]
      }
    }
  },
  "required": [
    "enumVal",
    "iface",
    "Details"
  ]
}
